ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -344,6 +431,22 @@ func (r *RayClusterReconciler) reconcilePods(instance *rayiov1alpha1.RayCluster)
 		if err := r.List(context.TODO(), &workerPods, client.InNamespace(instance.Namespace), filterLabels); err != nil {
 			return err
 		}
+
+		// delete the worker pod if it is marked unhealthy
+		for _, workerPod := range workerPods.Items {
+			if workerPod.Annotations == nil {
+				continue
+			}
+			if v, ok := workerPod.Annotations[common.RayNodeHealthStateAnnotationKey]; ok && v == common.PodUnhealthy {
+				if err := r.Delete(context.TODO(), &workerPod); err != nil {
```

Comment:
same question here. Why we still delete pod manually?

Reply:
when event reconcile mark the pod unhealthy, in pod reconcile loop we will delete the pod. In the future, we can do more fine-grained control of the readiness-probe-faild pod